webapp/src/modules/item/sagas.ts

```diff
@@ -76,6 +89,26 @@ export function* itemSaga(getIdentity: () => AuthIdentity | undefined) {
   yield takeEvery(SET_PURCHASE, handleSetItemPurchaseWithCard)
   yield takeEvery(FETCH_ITEM_REQUEST, handleFetchItemRequest)
 
+  // to avoid race conditions, just one fetch items request is handled at once in the browse page
+  function* handleOneFetchItemsRequestAtOnce(): SagaIterator {
```

Comment:
What if we parametrize this function? We could test it independently from the items actions as well.
```suggestion
  function* takeLatestByPath(actionType: string, path: string): SagaIterator {
```

Reply:
excellent idea! updated :)